src/compiler/crystal/semantic/literal_expander.cr

```diff
@@ -343,6 +343,14 @@ module Crystal
       end
     end
 
+    private def regex_new_call(node, value)
+      Call.new(Path.global("Regex").at(node), "new", value, regex_options(node)).at(node)
+    end
+
+    private def regex_options(node)
+      Call.new(Path.global(["Regex", "Options"]).at(node), "new", NumberLiteral.new(node.options.value.to_s).at(node)).at(node)
```

Comment:
The `to_s` is necessary to avoid a tight coupling between the value type of the compiler and stdlib. The base type of `Regex::Options` was changed to `UInt64` in #13223.